packages/caip/src/assetId/assetId.ts

```diff
@@ -145,18 +152,38 @@ export const fromAssetId: FromAssetId = assetId => {
 
   const chainId = toChainId({ chainNamespace, chainReference })
 
-  if (assetNamespace && assetReference && chainId) {
-    return {
-      chainId,
-      chainReference,
-      chainNamespace,
-      assetNamespace,
-      assetReference: ['erc20', 'erc721', 'bep20', 'bep721'].includes(assetNamespace)
-        ? assetReference.toLowerCase()
-        : assetReference,
+  const assetReferenceNormalized = (() => {
+    switch (assetNamespace) {
+      case 'erc20':
+      case 'bep20':
+      case 'erc721':
+      case 'erc1155':
+      case 'bep721':
+      case 'bep1155':
+        return assetReference.toLowerCase()
+      default:
+        return assetReference
```

Comment:
shouldn't the default error out in some way, so that we don't parse unsupported assets?

Reply:
Not sure, an "invalid" `assetNamespace` might be "invalid" from *our* heuristics point of view (unsupported by us) but "anything" can be an assetReference from a CAIP perspective (might not necessarily be a valid CAIP at a more granular level but would be syntactically correct, we slightly deviate from the CAIP spec at times either way i.e OSMO LP gamm assets)

Reply:
yeah the goal here is to avoid checksum addresses for evm chains, everything else we just treat as is at this time